tests/integration/streaming/test_streaming.py

```diff
@@ -11,22 +12,27 @@ async def task(self, doc: Document, **kwargs):
         for i in range(100):
             yield Document(text=f'{doc.text} {i}')
 
+    @requests(on='/world')
+    async def non_gen_task(self, doc: Document, **kwargs):
+        return Document(text=f'{doc.text}')
```

Comment:
but wait, how does this task work? this should raise an error now

Reply:
it is
<img width="754" alt="image" src="https://github.com/jina-ai/jina/assets/15269265/62b43e03-c99d-4a38-963a-6f0bb034007f">

Reply:
okey.

But I believe this should perhaps fail to start, not at client side.

Reply:
how would you know it should fail to start ? it is only an error because the client attempts to call it using streaming stubs

Reply:
u can analize in the `requests` decorator that it is a single Document annotated and is not a generator right?

Reply:
check and tests were added